### ISSUE TITLE
Bug: sort-object-types - Fix objects must not be overlapped in a report.

### ISSUE THREAD

**Issue by dylang** [azat-io/eslint-plugin-perfectionist#13]
### Describe the bug

New bug from v1.1.1:

```
Fix objects must not be overlapped in a report.
Occurred while linting /project/file-with-error.ts:10
Rule: "perfectionist/sort-object-types"
AssertionError [ERR_ASSERTION]: Fix objects must not be overlapped in a report.
    at mergeFixes (/project/node_modules/eslint/lib/linter/report-translator.js:152:9)
    at normalizeFixes (/project/node_modules/eslint/lib/linter/report-translator.js:182:16)
    at /project/node_modules/eslint/lib/linter/report-translator.js:349:49
    at Object.report (/project/node_modules/eslint/lib/linter/linter.js:1021:41)
    at /project/node_modules/eslint-plugin-perfectionist/dist/index.js:1:7540
    at I (/project/node_modules/eslint-plugin-perfectionist/dist/index.js:1:2732)
    at TSTypeLiteral (/project/node_modules/eslint-plugin-perfectionist/dist/index.js:1:7516)
    at /project/node_modules/eslint/lib/linter/timing.js:141:28
    at ruleErrorHandler (/project/node_modules/eslint/lib/linter/linter.js:1050:28)
    at /project/node_modules/eslint/lib/linter/safe-emitter.js:45:58
    at Array.forEach (<anonymous>)
    at Object.emit (/project/node_modules/eslint/lib/linter/safe-emitter.js:45:38)
    at NodeEventGenerator.applySelector (/project/node_modules/eslint/lib/linter/node-event-generator.js:297:26)
    at NodeEventGenerator.applySelectors (/project/node_modules/eslint/lib/linter/node-event-generator.js:326:22)
    at NodeEventGenerator.enterNode (/project/node_modules/eslint/lib/linter/node-event-generator.js:340:14)
    at CodePathAnalyzer.enterNode (/project/node_modules/eslint/lib/linter/code-path-analysis/code-path-analyzer.js:795:23)
    at /project/node_modules/eslint/lib/linter/linter.js:1085:32
    at Array.forEach (<anonymous>)
    at runRules (/project/node_modules/eslint/lib/linter/linter.js:1080:15)
    at Linter._verifyWithoutProcessors (/project/node_modules/eslint/lib/linter/linter.js:1330:31)
    at Linter._verifyWithConfigArray (/project/node_modules/eslint/lib/linter/linter.js:1706:21)
    at Linter.verify (/project/node_modules/eslint/lib/linter/linter.js:1412:65)
    at Linter.verifyAndFix (/project/node_modules/eslint/lib/linter/linter.js:1967:29)
    at verifyText (/project/node_modules/eslint/lib/cli-engine/cli-engine.js:245:48)
    at CLIEngine.executeOnFiles (/project/node_modules/eslint/lib/cli-engine/cli-engine.js:825:28)
    at ESLint.lintFiles (/project/node_modules/eslint/lib/eslint/eslint.js:551:23)
```

### Code example

Seems like the `// should just be string` is the cause.

```ts
export const generateListOfChanges = ({
    arrayOfChanges,
    propNameForWhatChanged,
    formatValue = (value) => `${value}%`
}: {
    arrayOfChanges: {
        isIncreased: boolean;
        isImproved: boolean;
        change: number;
        isSame: boolean;
        [propNameForWhatChanged: string]: boolean | number | string | unknown; // should just be string
    }[];
    propNameForWhatChanged: string;
    formatValue?: (value: number) => string;
}) => {
};
```

### ESLint version

v8.41.0

### ESLint Plugin Perfectionist version

v1.1.1

### Additional comments

Thanks for fixing these so quickly!

### Validations

- [X] Read the [docs](https://eslint-plugin-perfectionist.azat.io).
- [X] Check that there isn't [already an issue](https://github.com/azat-io/eslint-plugin-perfectionist/issues) that reports the same bug to avoid creating a duplicate.

**Comment by azat-io** [comment]
Thanks.  This seems to be a similar problem to #11
I will fix it soon.

**Comment by azat-io** [comment]
Fixed in 03e550820172c6cee52e686c5e694341cdbea327

Released in [v1.1.2](https://github.com/azat-io/eslint-plugin-perfectionist/releases/tag/v1.1.2)
